src/psyclone/tests/psyGen_test.py

```diff
@@ -2581,18 +2603,21 @@ def test_node_abstract_methods():
     with pytest.raises(NotImplementedError) as err:
         Node.view(loop)
     assert "BaseClass of a Node must implement the view method" in str(err)
+    Config.get().api = "dynamo0.3"
 
 
 def test_kern_ast():
     ''' Test that we can obtain the fparser2 AST of a kernel. '''
     from psyclone.gocean1p0 import GOKern
     from fparser.two import Fortran2003
+    Config.get().api = "gocean1.0"
```

Comment:
monkeypatch/fixture?

Reply:
@arporter , there are several places where I set a default api in a test function, mostly in 'generic' (non-api-specific) tests like alggen_test etc. By now in nearly all places this is not necessary for the tests anymore. Short explanation:
Originally I used Config.get().api_conf() in dynamo* and gocean*. For this to work I indeed needed to change the default API in the tests to make sure the right api-specific config will be picked up. But then I realised that in e.g. dynamo0p3 I should always get the dynamo0.3 config, so I changed the code to use: config.get().api_conf("dynamo0.3"). Since the API-files (dynamo*, gocean*) are responsible for translating the api-specific strings to the AccessType, there is no need to have a default API specified ... unless there is an error. In case of an error (especially if detected by a baseclass e.g. in psygen) a default API is needed to translate AccessType back to the api-specific string.

As summary: in all test cases that do not trigger an (AccessType-related) error, there is no need to change the API anymore, only the tests which test (AccessType-related) exception handling need a default being defined in order for the right error message.

So I could basically remove nearly all settings of an API.  The fixture in each API-file still makes sense (since e.g. a gocean1p0 test will likely test gocean1p0 related errors).

So would you be ok if I just remove the unnecessary settings of the api (with the only exception being tests that raise accesstype-related errors) - and since we have a fixture, that would only leave some non-API-specific tests (psyGen_test, alggen_test) where it is actually required.

Reply:
Yes, that sounds good to me!

Reply:
OK, I changed this everywhere, I've added 'outdated' to all the corresponding comments here.

Reply:
Done.